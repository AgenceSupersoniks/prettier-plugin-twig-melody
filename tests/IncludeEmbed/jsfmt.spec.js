run_spec(__dirname, ["melody"], { twigFollowOfficialCodingStandards: false });
