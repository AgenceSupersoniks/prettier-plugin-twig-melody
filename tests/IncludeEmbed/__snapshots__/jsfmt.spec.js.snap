// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`block.melody.twig 1`] = `
<section key="0">
{% block hello %}
    <div key="1" class="test">
        <span>Hello</span>
    </div>
{% endblock %}
</section>

{% block bar foo %}

{{ block('hello') }}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<section key="0">
    {% block hello %}
        <div key="1" class="test">
            <span>Hello</span>
        </div>
    {% endblock hello %}
</section>

{% block bar foo %}

{{ block("hello") }}
`;

exports[`embed.melody.twig 1`] = `
{% extends "parent.twig" %}

{% block hello %}
    <div class="test-{{ foo }} bar">
        {% embed "foo.twig" with { foo: 'bar' } %}
            {% block hello %}
                {{ fun }}
                {% embed "bar.twig" %}
                    {% block hello %}
                        {{ message }}
                    {% endblock %}
                    {% block test %}
                    {% endblock %}
                {% endembed %}
            {% endblock hello %}
        {% endembed %}
    </div>
{% endblock %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% extends "parent.twig" %}

{% block hello %}
    <div class="{{ "test-" ~ foo ~ " bar" }}">
        {% embed "foo.twig" with { "foo": "bar" } %}
            {% block hello %}
                {{ fun }}
                {% embed "bar.twig" %}
                    {% block hello %}
                        {{ message }}
                    {% endblock hello %}
                    {% block test %}

                    {% endblock test %}
                {% endembed %}
            {% endblock hello %}
        {% endembed %}
    </div>
{% endblock hello %}
`;

exports[`extendsEmbed.melody.twig 1`] = `
{% extends "parent.twig" %}
{% extends someVar %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% extends "parent.twig" %}
{% extends someVar %}
`;

exports[`import.melody.twig 1`] = `
{% import "forms.html" as forms %}
{% import "aVeryLongAndConvolutedAndIntertwinedFilename.html" as someQuiteEccentricLocalVariableName %}

{% from "macros.twig" import hello %}
{% from 'forms.html' import input as input_field,textarea %}
{% from 'aVeryLongAndConvolutedAndIntertwinedFilename.html' import input as input_field, textarea, select, password as pw_field, radioButton %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% import "forms.html" as forms %}
{% import "aVeryLongAndConvolutedAndIntertwinedFilename.html"
    as someQuiteEccentricLocalVariableName
%}

{% from "macros.twig" import hello %}
{% from "forms.html" import input as input_field, textarea %}
{% from "aVeryLongAndConvolutedAndIntertwinedFilename.html" import
    input as input_field,
    textarea,
    select,
    password as pw_field,
    radioButton
%}
`;

exports[`include.melody.twig 1`] = `
<div key="1" class="test">
    {{ message | lower | upper }}{% flush %}
    <span class="">{{ _context.name[1:] }}</span>
    {{ block('test') }}
    {{ include('test.twig') }}
    {% include 'test.twig' %}
</div>

{% include './Star.twig' only %}

{% include "./Flag.twig" with {
"styleModifier": flagModifiers,
"dataVariables": dataVariables,
"text": "ie_topdeal"
} only %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
<div key="1" class="test">
    {{ message | lower | upper }}{% flush %}
    <span class="">{{ _context.name[1:] }}</span>
    {{ block("test") }}
    {{ include("test.twig") }}
    {% include "test.twig" %}
</div>

{% include "./Star.twig" only %}

{% include "./Flag.twig" with {
    "styleModifier": flagModifiers,
    "dataVariables": dataVariables,
    "text": "ie_topdeal"
} only %}
`;

exports[`mount.melody.twig 1`] = `
{% mount './component' as 'bar' %}

{% mount async './parts/#{ part }.twig' as 'bar-#{part}' with {foo: 'bar'} delay placeholder by 1s %}
  Loading...
{% catch err %}
  Failed to load with {{ err }}
{% endmount %}

<div class="profile-wrapper account">
    {% if isProfilePageDisplayed %}
        {% mount '../Profile' as 'profile' with {
            user: user,
            expressBookingIframeUrl: expressBookingIframeUrl,
            paymentMethodsIframeUrl: paymentMethodsIframeUrl,
            isAboveScreenMedium: isAboveScreenMedium,
            anchor: anchor,
            recentlyViewedExpanded: recentlyViewedExpanded
        } %}
    {% endif %}
</div>
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% mount "./component" as "bar" %}

{% mount async "./parts/" ~ part ~ ".twig" as "bar-" ~ part with {
        "foo": "bar"
    }
    delay placeholder by 1s
%}
    Loading...
{% catch err %}
    Failed to load with {{ err }}
{% endmount %}

<div class="profile-wrapper account">
    {% if isProfilePageDisplayed %}
        {% mount "../Profile" as "profile" with {
                "user": user,
                "expressBookingIframeUrl": expressBookingIframeUrl,
                "paymentMethodsIframeUrl": paymentMethodsIframeUrl,
                "isAboveScreenMedium": isAboveScreenMedium,
                "anchor": anchor,
                "recentlyViewedExpanded": recentlyViewedExpanded
            }
        %}
    {% endif %}
</div>
`;

exports[`useStatement.melody.twig 1`] = `
{% use "foo.twig" %}

{% use "blocks.html" with sidebar as base_sidebar, title as base_title %}

{% use "extraLongNameBlocks.html" with sidebar as base_sidebar, title as base_title %}
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
{% use "foo.twig" %}

{% use "blocks.html" with sidebar as base_sidebar, title as base_title %}

{% use "extraLongNameBlocks.html" with
    sidebar as base_sidebar,
    title as base_title
%}
`;
